lib/semian.rb

```diff
@@ -52,38 +53,52 @@
 #
 # This is the same as the previous example, but overrides the timeout from the default value of 500 milliseconds to 1 second.
 #
-class Semian
+module Semian
+  extend self
+
   class << self
-    # Registers a resource.
-    #
-    # +name+: Name of the resource - this can be either a string or symbol.
-    #
-    # +tickets+: Number of tickets. If this value is 0, the ticket count will not be set,
-    # but the resource must have been previously registered otherwise an error will be raised.
-    #
-    # +permissions+: Octal permissions of the resource.
-    #
-    # +timeout+: Default timeout in seconds.
-    #
-    # Returns the registered resource.
-    def register(name, tickets: 0, permissions: 0660, timeout: 1)
-      resource = Resource.new(name, tickets, permissions, timeout)
-      resources[name] = resource
-    end
+    attr_accessor :logger
+  end
+
+  self.logger = Logger.new(nil)
 
-    # Retrieves a resource by name.
-    def [](name)
-      resources[name]
-    end
+  # Registers a resource.
```

Comment:
This is a little vague since we both create a Semian resource and Circuit Breaker